src/webgpu/shader/types.ts

```diff
@@ -36,3 +42,146 @@ export const kMatrixContainerTypeInfo = /* prettier-ignore */ {
 } as const;
 /** List of all matNxN<> container types. */
 export const kMatrixContainerTypes = keysOf(kMatrixContainerTypeInfo);
+
+/**
+ * Generate a bunch types (vec, mat, sized/unsized array) for testing.
+ */
+export function* generateTypes({
+  storageClass,
+  baseType,
+  containerType,
+  supportsAtomics = false,
+}: {
+  storageClass: string;
+  /** Base scalar type (i32/u32/f32/bool). */
+  baseType: ScalarType;
+  /** Container type (scalar/vector/matrix/array) */
+  containerType: ContainerType;
+  /** Whether to wrap the baseType in `atomic<>`. */
+  supportsAtomics?: boolean;
```

Comment:
nit: maybe just "atomic". We're indicating whether we want it to be atomic, not whether it is atomic

Reply:
Yes, that makes sense. How about isAtomic?